src/components/App/App.js

```diff
@@ -26,12 +31,37 @@ const PAGES_TO_SHOW_TIMEOUT = [
 ];
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    initFirebase();
+    this.firebaseAuth = firebase.auth();
+    window.auth = this.firebaseAuth;
+  }
+
   state = {
     showTimer: false,
-    isOnline: true
+    isOnline: true,
+    loggedIn: null
+  };
+
+  firebaseUiConfig = {
+    signInOptions: [
+      firebase.auth.GoogleAuthProvider.PROVIDER_ID,
+      {
+        provider: firebase.auth.EmailAuthProvider.PROVIDER_ID,
+        requireDisplayName: false
+      }
+    ],
+    callbacks: {
+      signInSuccessWithAuthResult: res => this.setState({loggedIn: !!res.user})
```

Comment:
The `!!` is pretty nifty but can be a bit confusing/hard to read for some, could we cast to a boolean using `Boolean` to make this clearer?

```
this.setState({ loggedIn: Boolean(res.user) })
```

Or, if we know what the falsy value would be and want to be more explicit

```
this.setState({ loggedIn: res.user !== undefined })
```